app/javascript/packs/form-steps-wait.tsx

```diff
@@ -116,22 +114,19 @@ export class FormStepsWait {
     const response = await window.fetch(action, {
       method,
       body: new window.FormData(form),
+      headers: {
+        Accept: 'application/json',
+      },
     });
 
-    if ('polyfill' in window.fetch) {
-      // The fetch polyfill is implemented using XMLHttpRequest, which suffers from an issue where a
-      // Content-Type header from a POST is carried into a redirected GET, which is exactly the flow
-      // we are handling here. The current version of Rack neither handles nor provides easy insight
-      // into an empty-bodied (GET) multi-part form. This will change in Rack 3 with the addition of
-      // the Rack::Multipart::EmptyContentError class. In the meantime, only allow non-polyfilled
-      // fetch environmnents to handle the initial response.
-      //
-      // See: https://github.com/whatwg/fetch/issues/609
-      // See: https://github.com/rack/rack/issues/1603
-      this.poll();
-    } else {
-      this.handleResponse(response);
-    }
+    const channelDetails = await response.json();
+
+    this.consumer.subscriptions.create(
+      { channel: 'DocumentCaptureResultChannel', ...channelDetails },
+      {
+        received: () => this.poll(),
```

Comment:
re-using `this.poll()` is just a demo to minimize code changes, right? If we did choose to go fully in on websockets we wouldn't need to poll at all right?

Reply:
> re-using `this.poll()` is just a demo to minimize code changes, right? If we did choose to go fully in on websockets we wouldn't need to poll at all right?

Correct. More detail in the original comment:

>The implementation could also be further optimized. Currently, the channel is used to let the client know that a result is ready, but the client still needs to send a request to retrieve the result. A more optimized implementation would include the result in the initial message.

As-is, this is mostly intended as an MVP to provide the same baseline functionality, and we could iterate to optimize the implementation as a follow-on.